plugin/src/main/java/org/autorefactor/refactoring/rules/BooleanRefactoring.java

```diff
@@ -433,11 +426,33 @@ private Expression getReturnExpression(MethodDeclaration md, Expression ifCondit
                         + md.getName().getIdentifier() + ", but found " + qualifiedName);
     }
 
-    private Expression maybeGetExpression(String expressionName, Expression ifCondition, Boolean thenBoolLiteral,
-            Boolean elseBoolLiteral) {
-        if (areOppositeValues(thenBoolLiteral, elseBoolLiteral)) {
+    private Expression maybeGetExpression(String expressionName, Expression ifCondition, Expression thenExpression,
+            Expression elseExpression) {
+        if (areOppositeBooleanValues(thenExpression, elseExpression)) {
             final Name booleanName = getBooleanName(ifCondition);
-            return getExpression(thenBoolLiteral, ifCondition, expressionName, booleanName);
+            return getExpression(getBooleanLiteral(thenExpression), ifCondition, expressionName, booleanName);
+        } else if (isPrimitive(thenExpression) || isPrimitive(elseExpression)) {
+            // If both expressions are primitive, there can't be null pointer
+            // If only one expression is primitive, null pointer is already
+            // possible so we don't care
+            if (getBooleanLiteral(thenExpression) != null && getBooleanLiteral(elseExpression) == null) {
+
+                if (getBooleanLiteral(thenExpression)) {
+                    return b.infixExpr(b.copy(ifCondition), InfixExpression.Operator.CONDITIONAL_OR,
+                            b.copy(elseExpression));
+                } else {
+                    return b.infixExpr(b.negate(ifCondition, Copy.COPY), InfixExpression.Operator.CONDITIONAL_AND,

```

Comment:
Please use static imports for `InfixExpression.Operator.CONDITIONAL_OR` and `InfixExpression.Operator.CONDITIONAL_AND` in this whole class.